src/util/pubkey_finder/email_sigs_gcd.py

```diff
@@ -42,13 +53,106 @@ async def has_known_keys(prisma: Prisma, dsp: Dsp, dspsWithKnownKeys: set[Dsp])
 		return True
 	return False
 
+async def validate_signature(keyData: str, sig: EmailSignature) -> bool:
+    try:
+        # Try a few different approaches to verify the signature
+        rsa_key = RSA.import_key(binascii.a2b_base64(keyData))
+        verifier = PKCS1_v1_5.new(rsa_key)
+        sig_bytes = binascii.a2b_base64(sig.dkimSignature)
+        logging.info(f'validating signature {sig.id} with message {sig.headerHash} and signature {sig.dkimSignature} with sig bytes {sig_bytes} with key {keyData}')
+        logging.info(f'Signature bytes length: {len(sig_bytes)}')
+ 
+        # Approach 1: Using raw hash bytes
+        h = SHA256.new()
+        h.digest = lambda: bytes.fromhex(sig.headerHash)
+        if verifier.verify(h, sig_bytes):
+            logging.info(f'signature {sig.id} validated with raw hash bytes')
+            return True
+            
+        # Approach 2: Using pkcs1 padding like in gcd_solver
+        size_bytes = len(sig_bytes)
+        message, signature = message_sig_pair(size_bytes, sig.headerHash, sig_bytes, 'sha256')
+
+        decrypted = pow(signature, rsa_key.e, rsa_key.n)
+        logging.info(f'Approach 2 comparison:')
+        logging.info(f'Expected: {hex(message)}')
+        logging.info(f'Got:      {hex(decrypted)}')
+        
+        if pow(signature, rsa_key.e, rsa_key.n) == message:
```

Comment:
This fails and am not sure why. Calling RSA verification directly also fails. I didn't understand how the bytes just don't match